File: src/main/java/br/com/jiratorio/domain/request/UpdateBoardRequest.java
package br.com.jiratorio.domain.request;

import br.com.jiratorio.domain.DueDateType;
import br.com.jiratorio.domain.impediment.ImpedimentType;
import br.com.jiratorio.domain.DynamicFieldConfig;
import java.util.List;
import lombok.Data;

@Data
public class UpdateBoardRequest {

    private String name;

    private String startColumn;

    private String endColumn;

    private List<String> fluxColumn;

    private List<String> ignoreIssueType;

    private String epicCF;

    private String estimateCF;

    private String systemCF;

    private String projectCF;

    private Boolean ignoreWeekend;

    private ImpedimentType impedimentType;

    private List<String> impedimentColumns;

    private List<DynamicFieldConfig> dynamicFields;

    private List<String> touchingColumns;

    private List<String> waitingColumns;

    private String dueDateCF;

    private DueDateType dueDateType;

}

File: src/main/java/br/com/jiratorio/domain/request/UpdateBoardRequest.kt
package br.com.jiratorio.domain.request

import br.com.jiratorio.domain.DueDateType
import br.com.jiratorio.domain.DynamicFieldConfig
import br.com.jiratorio.domain.impediment.ImpedimentType

class UpdateBoardRequest {

    var name: String? = null

    var startColumn: String? = null

    var endColumn: String? = null

    var fluxColumn: MutableList<String>? = null

    var ignoreIssueType: MutableList<String>? = null

    var epicCF: String? = null

    var estimateCF: String? = null

    var systemCF: String? = null

    var projectCF: String? = null

    var ignoreWeekend: Boolean? = null

    var impedimentType: ImpedimentType? = null

    var impedimentColumns: MutableList<String>? = null

    var dynamicFields: MutableList<DynamicFieldConfig>? = null

    var touchingColumns: MutableList<String>? = null

    var waitingColumns: MutableList<String>? = null

    var dueDateCF: String? = null

    var dueDateType: DueDateType? = null

}
